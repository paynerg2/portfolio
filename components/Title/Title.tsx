import { Box, Heading, useColorModeValue } from '@chakra-ui/react';
import { useEffect, useState } from 'react';
import { AnimatedText } from '../AnimatedText/AnimatedText';

const descriptors = ['Web Developer', 'Math Enthusiast', 'Science Fiction Fan'];

const Title = () => {
    const [index, setIndex] = useState(0);
    const highlightColor = useColorModeValue('blue.500', 'blue.200');

    useEffect(() => {
        const modulus = descriptors.length;
        setTimeout(() => {
            setIndex((index + 1) % modulus);
        }, 6000);
    });

    return (
        <Box>
            <Box h="1.5em">
                <AnimatedText
                    animationType={index}
                    color={highlightColor}
                    fontSize="xl"
                    fontWeight="semibold"
                    text={descriptors[index]}
                />
            </Box>
            <Heading size="4xl" m={0}>
                Rodney Payne
            </Heading>
        </Box>
    );
};

export default Title;
