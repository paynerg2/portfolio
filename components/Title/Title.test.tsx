import { render } from '@testing-library/react';
import Title from './Title';

const mock = function () {
    return {
        observe: jest.fn(),
        unobserve: jest.fn(),
        disconnect: jest.fn(),
    };
};
window.IntersectionObserver = jest.fn().mockImplementation(mock);

describe('Title Component', () => {
    it('renders the component', () => {
        const component = render(<Title />);
        expect(component).toMatchSnapshot();
    });
});
