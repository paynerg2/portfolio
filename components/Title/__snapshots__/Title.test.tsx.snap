// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-0"
      >
        <div
          class="css-1sknhd9"
        >
          <div
            aria-label="Web Developer"
            class="chakra-container css-1o2ugtj"
          >
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              W
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              e
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              b
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
               
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              D
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              e
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              v
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              e
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              l
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              o
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              p
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              e
            </span>
            <span
              style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              r
            </span>
             
          </div>
        </div>
        <h2
          class="chakra-heading css-dpoqzr"
        >
          Rodney Payne
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-0"
    >
      <div
        class="css-1sknhd9"
      >
        <div
          aria-label="Web Developer"
          class="chakra-container css-1o2ugtj"
        >
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            W
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            e
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            b
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
             
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            D
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            e
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            v
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            e
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            l
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            o
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            p
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            e
          </span>
          <span
            style="position: relative; display: inline-block; width: auto; opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            r
          </span>
           
        </div>
      </div>
      <h2
        class="chakra-heading css-dpoqzr"
      >
        Rodney Payne
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
